tests/Hamcrest/File/IsFileWithCanonicalPathTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/**
+ * @author Vasek Brychta <vaclav@brychtovi.cz>
+ */
+
+namespace Hamcrest\File;
+
+use Hamcrest\AbstractMatcherTest;
+
+class IsFileWithCanonicalPathTest extends AbstractMatcherTest
+{
+	private static $CANONICAL_PATH;
+	private static $FILENAME;
+	private static $FILE_PATH;
+	private static $FILE;
+
+	private static $DIFFERENT_CANONICAL_PATH;
+
+	/**
+	 * @beforeClass
+	 */
+	public static function initializeFiles()
+	{
+		self::$CANONICAL_PATH = __FILE__;
+		self::$FILENAME = pathinfo(__FILE__, PATHINFO_BASENAME);
+		self::$FILE_PATH = __DIR__ . '/./' . self::$FILENAME;
+		self::$FILE = new \SplFileInfo(self::$FILE_PATH);
+
+		self::$DIFFERENT_CANONICAL_PATH = __DIR__ . '/different-file.txt';
+	}
+
+	protected function createMatcher()
+	{
+		return IsFileWithCanonicalPath::aFileWithCanonicalPath(not(anything()));
```

Comment:
Something isn't right here. Should the `createMatcher` method have an argument given to the matcher and then all tests in this class create matcher via the `createMatcher` method?

Please see how it's implemented in other matcher tests, that have arguments.